spec/board_spec.rb

```diff
@@ -100,5 +100,103 @@
       expect(@board.valid_placement?(@cruiser, ['A2', 'B2', 'C2'])).to be(false)
     end
   end
+  
+  describe '#render' do
+    it 'can render a board without boats' do
+      expect(@board.render).to eq("  1 2 3 4 \nA . . . . \nB . . . . \nC . . . . \nD . . . . \n")
+    end
+
+    it 'can render a boat' do
+      expect(@board.render).to eq("  1 2 3 4 \nA . . . . \nB . . . . \nC . . . . \nD . . . . \n")
+      @board.place(@cruiser, ['A1', 'A2', 'A3'])
+      cell_1 = @board.cells['A1']
+      cell_2 = @board.cells['A2']
+      cell_3 = @board.cells['A3']
+      expect(cell_1.empty?).to eq(false)
+      expect(cell_2.empty?).to eq(false)
+      expect(cell_3.empty?).to eq(false)
+      expect(@board.render(true)).to eq("  1 2 3 4 \nA S S S . \nB . . . . \nC . . . . \nD . . . . \n")
+    end
+
+    it 'can render a boat and render a miss' do
+      expect(@board.render).to eq("  1 2 3 4 \nA . . . . \nB . . . . \nC . . . . \nD . . . . \n")
+      @board.place(@cruiser, ['A1', 'A2', 'A3'])
+      cell_1 = @board.cells['A1']
```

Comment:
Again lines 124-126, 128-130 seem unnecessary? I think leaving cell_4 in there is fine (you could also just replace line 132 with `@board.cells['A4'].fire_upon`  and, again, I think lines 133-135 are kinda redundant because when you .render in line 136, it essentially shows the same thing as lines 133-135.